li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -131,12 +132,12 @@
 
   // A counter of the threads in the middle of sending messages. This is needed to ensure when we close the producer
   // everything is audited.
-  private final AtomicInteger _numThreadsInSend;
+  private final LongAdder _numThreadsInSend = new LongAdder();
   private volatile boolean _closed;
 
   // This is null if the underlying producer does not have an implementation for time-bounded flush
   private final Method _boundedFlushMethod;
-  private final AtomicInteger _boundFlushThreadCount = new AtomicInteger();
+  private final LongAdder _boundFlushThreadCount = new LongAdder();
```

Comment:
This patch replaces `AtomicInteger` with `LongAdder` for `_numThreadsInSend` and `_boundFlushThreadCount`. I can see how there might be some higher contention for `_numThreadsInSend` and that it might benefit from switching to a `LongAdder` type. I don't think we can say the same for `_boundFlushThreadCount` because:
- It is unlikely that there are many user threads concurrently trying to flush the producer. 
- It seems like we are reading the `intValue` of the long adder `_boundFlushThreadCount` to create a unique identifier for the flush thread. I think that is pretty expensive and not guaranteed to return accurate result under concurrent updates https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/LongAdder.html#sum--

Reply:
Good point on LongAdder not providing an atomic snapshot. I agree that it should not be used for unique number generation. Javadoc on LongAdder does say that it is preferred for updating a common sum, but not for fine-grained synchronization control. I will revert both to AtomicInteger for now.